### PR TITLE
feat(forge): constructor-args-path option for verify-contract

### PR DESCRIPTION
**Pull request by 0xYYY**
## Motivation

Closes https://github.com/foundry-rs/foundry/issues/3065.

## Solution

Extract `read_constructor_args_file` from `create` as a util function and reuse it in `verify-contract`.

### CODE REVIEW

File: cli/src/cmd/utils.rs [hunk 1]
```diff
@@ -269,3 +270,24 @@ where
         config
     }
 }
+
+pub fn read_constructor_args_file(constructor_args_path: PathBuf) -> eyre::Result<Vec<String>> {
+    if !constructor_args_path.exists() {
+        eyre::bail!("Constructor args file \"{}\" not found", constructor_args_path.display());
+    }
+    let args = if constructor_args_path.extension() == Some(std::ffi::OsStr::new("json")) {
+        read_json_file(&constructor_args_path).map_err(|e| {
```
Comment (mattsse) [inline cli/src/cmd/utils.rs:279]:
```suggestion
        read_json_file(&constructor_args_path).wrap_err(|e| {
```

File: cli/src/cmd/utils.rs [hunk 2]
```diff
@@ -269,3 +270,24 @@ where
         config
     }
 }
+
```
Comment (mattsse) [inline cli/src/cmd/utils.rs:273]: brief docs would be nice!

File: cli/src/cmd/utils.rs [hunk 3]
```diff
@@ -269,3 +270,24 @@ where
...
+            eyre::eyre!(
+                "Constructor args file \"{}\" must encode a json array: \"{}\"",
+                constructor_args_path.display(),
+                e
+            )
+        })?
+    } else {
+        fs::read_to_string(constructor_args_path)?
+            .split_whitespace()
+            .map(str::to_string)
```
Comment (mattsse) [inline cli/src/cmd/utils.rs:289]: don't think converting al items to String first is required?

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome, ty!

extracting this as standalone functions is a good decision, but I noticed a few things that there are some things that were in there before this PR, but that we should change.

File: cli/src/cmd/utils.rs (hunk 3 above)
Comment (0xYYY) [reply cli/src/cmd/utils.rs]: i think `Vec<String>` can't be built from `std::iter::Iterator<Item=&str>` that `split_whitespace()` returns

Comment (mattsse) [reply cli/src/cmd/utils.rs]: ah right, this is Vec<>, nvm then

**Review by mattsse — APPROVED** [review]

**Comment by 0xYYY** [comment]
realized that there's an issue in this implementation: `VerifyContract.constructor_arguments` actually takes **ABI-encoded** args as input, but currently we are using whitespace separated args from file

will send another PR to fix this

**Pull request merged**